src/inmanta/server/services/compilerservice.py

```diff
@@ -601,13 +602,104 @@ async def request_recompile(
         Recompile an environment in a different thread and taking wait time into account.
 
         :param notify_failed_compile: if set to True, errors during compilation will be notified using the
-        "failed_compile_message".
-        if set to false, nothing will be notified. If not set then the default notifications are
-        sent (failed pull stage and errors during the do_export)
+                                      "failed_compile_message".
+                                      if set to false, nothing will be notified. If not set then the default notifications are
+                                      sent (failed pull stage and errors during the do_export)
         :param failed_compile_message: the message used in notifications if notify_failed_compile is set to True.
         :return: the compile id of the requested compile and any warnings produced during the request
+        """
+        return await self._request_recompile(
+            env,
+            force_update,
+            do_export,
+            remote_id,
+            metadata,
+            env_vars,
+            partial,
+            removed_resource_sets,
+            exporter_plugin,
+            notify_failed_compile,
+            failed_compile_message,
+        )
+
+    async def request_recompile_in_transaction(
```

Comment:
Why a separate method instead of an optional parameter?

Reply:
I wonder as well.

Reply:
I wanted to make it sufficiently clear that this call would require the notify, but maybe the presence of the `in_db_transaction` argument is sufficient. I will change it.